src/components/Menu/Menu.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import './Menu.css';
+import MenuLinks from './MenuLinks';
+import { uniqueNoOfDishes, getDish } from './MenuHelper';
+import Dishes from '../Dishes';
+import Cart from '../Cart';
+class Menu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dishes: {},
+      quantities: [],
+      noOfDishes: 0,
```

Comment:
the **noOfDishes** state is dependent on **dishes** state, so it will only change when the dishes state changes,
and you are only using it in clear cart to initialize a new array, instead you can remove this state and derive a variable from length of dishes state and use it when required in clear cart or if the data is too large then you can have a length property for dishes state and use it.  by this you will have a single state and it will simpler to handle or debug

Reply:
Length can't be calculated directly because there are duplicates like the same dishes are there in recommended and some other categories so unique dishes are needed to calculate. To avoid repetition I stored it in state.